nums/core/array/blockarray.py

```diff
@@ -410,14 +410,59 @@ def reduce_axis(self, op_name, axis, keepdims=False):
 
     def __matmul__(self, other):
         if len(self.shape) > 2:
-            return self.tensordot(other, 2)
+            # TODO: (bcp) NumPy's implementation does a stacked matmul, which is not supported yet.
+            raise NotImplementedError("Matrix multiply for tensors of rank > 2 not supported yet.")
         else:
             return self.tensordot(other, 1)
 
     def tensordot(self, other, axes=2):
         if not isinstance(other, BlockArray):
             raise ValueError("Cannot automatically construct BlockArray for tensor operations.")
 
+        # Error checking before everything gets passed into BlockArray operations. Modified from the
+        # original NumPy tensordot method for error checking:
+        # https://github.com/numpy/numpy/blob/v1.20.0/numpy/core/numeric.py#L949-L1139
+        try:
+            iter(axes)
```

Comment:
Can we move this entire try/except sequence of tests into a function in `nums/core/array/utils.py`? Pass in the parameters needed to run the test, and name it `np_tensordot_param_test`.

Call `np_tensordot_param_test` from here, and before calling `np_tensordot_param_test`, test to ensure `axes` is integer, because currently that's all we support.